inspire_utils/name.py

```diff
@@ -173,17 +173,19 @@ def _is_roman_numeral(suffix):
 
         # Create first and middle
         first_name = _ensure_dotted_initials(self.first)
-        middle_name = _ensure_dotted_initials(self.middle)
-
-        if _is_initial(first_name) and _is_initial(middle_name):
-            normalized_names = u'{first_name}{middle_name}'
+        middle_names = [_ensure_dotted_initials(name) for name in self.middle_list]
```

Comment:
you shouldn't do this only for middle names,
```
import itertools
first_and_middle_names = (_ensure_dotted_initials(name) for name in chain(self.first_list, self.middle_list))
```
then you don't have to special case first vs middle names.